src/core/validate-graph.ts

```diff
@@ -116,15 +112,37 @@ export function validatePackageURL(pkg: types.PkgInfo): void {
       }
 
       default:
+        assert(pkg.name === purl.name, `name and packageURL name do not match`);
+    }
+
+    // validate package version
+    switch (purl.type) {
+      // the Snyk version of a golang module is either
+      // - the version without "v", e.g. v1.2.3 -> 1.2.3
+      // - the pseudo-version hash, e.g. v0.0.0-000-acf48ae230a1 -> #acf48ae230a1
+      case 'golang': {
+        let version = purl.version;
+        const maybePseudoVersion =
+          purl.version && reGolangPseudoVersion.exec(purl.version);
```

Comment:
This feels unsafe to me. If `purl.version` is truthy and the regex.exec is falsey, the value of `maybePseudoVersion` will end up being `purl.version` and we'll fall into the first `if` below and turn version into the 5th character of the version (instead of the regex match)

Reply:
nice! Did that result in any new tests?

Reply:
It did! I added an additional test case for "incompatible" versions (they end in `+incompatible`).